cli/src/commands/config.rs

```diff
@@ -270,3 +284,14 @@ pub(crate) fn cmd_config_edit(
     let config_path = get_new_config_file_path(&args.config_args.get_source_kind(), command)?;
     run_ui_editor(command.settings(), &config_path)
 }
+
+#[instrument(skip_all)]
+pub(crate) fn cmd_config_path(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &ConfigPathArgs,
+) -> Result<(), CommandError> {
+    let config_path = get_new_config_file_path(&args.config_args.get_source_kind(), command)?;
+    writeln!(ui.stdout(), "{}", config_path.display())?;
```

Comment:
nit: `cmd_workspace_root()` errors out on invalid utf-8 path. maybe it's better than printing lossy data.

Reply:
I guess it shouldn't be a part of `get_new_config_file_path` because the path might not be UTF-8 but still be openable?

One thing I don't like about cmd_workspace_root is that the error provides no context about what the path should be, maybe we should add `path.display()` to the error. (I'm assuming it's the lossy function you meant). Maybe I'll make a `write_path` helper.

Reply:
Separate question: will this output be wrapped to the terminal width or passed through a pager? If so, is there an easy way to disable that?

Reply:
> I guess it shouldn't be a part of `get_new_config_file_path` because the path might not be UTF-8 but still be openable?

Probably no. Caller needs to do some conversion anyway if the return type is `PathBuf`.

> One thing I don't like about cmd_workspace_root is that the error provides no context about what the path should be, maybe we should add `path.display()` to the error. (I'm assuming it's the lossy function you meant). Maybe I'll make a `write_path` helper.

I don't care much about the error because that would be rare. If we're serious, the helper function can print bytes on Unix.

Reply:
Not sure if you missed it; any thoughts about the "separate question" above, https://github.com/martinvonz/jj/pull/2907#discussion_r1475404046? I'm guessing it's fine already, but wanted to double-check.

Reply:
> will this output be wrapped to the terminal width or passed through a pager?

jj wouldn't spawn pager, but it's up user to pipe the output to pager, and I don't think we should control that.

Reply:
More importantly, it's not wrapped, right?

I'll mark this as resolved: I fixed your original comment (but was too lazy to make a helper function), and I'm 90% sure the answer to the above question is "right".